### PR TITLE
add new resource list page

### LINKED ISSUE #4314: Page that explains and enumerates all the node / resource types
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

We can add more detail here:
https://docs.getdbt.com/reference/global-configs/resource-type

Alternatively, we could create a new stand-alone page.

### What part(s) of the page would you like to see updated?

The closest documentation to enumerating the [valid resource types](https://github.com/dbt-labs/dbt-core/blob/a1f64510907cab3470060fe2eb270e007a9a2522/core/dbt/artifacts/resources/types.py#L18-L37) within dbt is [this](https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#affected-states-by-node-type).

Without a single page that enumerates all the resource types, it makes it harder to do documentation like is needed for https://github.com/dbt-labs/docs.getdbt.com/issues/4313.

It would be nice if our documentation also makes the distinction between the types that can be [versioned, ref'd, and/or executed](https://github.com/dbt-labs/dbt-core/blob/a1f64510907cab3470060fe2eb270e007a9a2522/core/dbt/node_types.py#L6-L26)

### dbt project containing "all" resource types

Here's a simple dbt project that has quite a few different resource types (but probably not all): https://github.com/dbt-labs/docs.getdbt.com/issues/4492#issuecomment-1881658603

### Main resource types

```python
    Model = "model"  # executable / refable / versioned
    Seed = "seed"  # executable / refable
    Snapshot = "snapshot"  # executable / refable
    Analysis = "analysis"  # executable
    Source = "source"
```

### Testing-related resource types

```python
    Test = "test"  # executable
    Unit = "unit_test"
    Fixture = "fixture"
```

### Semantic layer resource types

```python
    SemanticModel = "semantic_model"
    Metric = "metric"
    SavedQuery = "saved_query"
```

### Other resource types

```python
    Exposure = "exposure"
    Macro = "macro"
    Group = "group"
```

```python
    Operation = "operation"  # executable
    RPCCall = "rpc"  # executable
    SqlOperation = "sql_operation"  # executable
    Documentation = "doc"  # executable
```

### Potential table

See below for a re-imagining of the table [here](https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#affected-states-by-node-type).

> [!WARNING]  
> The following information hasn't been verified for completeness or correctness!

|            Node           | Executed in DAG | Exists in database | Created by execution | Upstream lineage | Downstream lineage |
|:-------------------------:|:---------------:|:------------------:|:--------------------:|:----------------:|:------------------:|
| Source                    |                 |          ✅         |                      |                  |          →         |
| Snapshot                  |        ✅        |          ✅         |           ✅          |         ←        |          →         |
| Model - incremental       |        ✅        |          ✅         |           ✅          |         ←        |          →         |
| Model - materialized view |        ✅        |          ✅         |           ✅          |         ←        |          →         |
| Model - view              |        ✅        |          ✅         |           ✅          |         ←        |          →         |
| Model - table             |        ✅        |          ✅         |           ✅          |         ←        |          →         |
| Seed                      |        ✅        |          ✅         |           ✅          |                  |          →         |
| Model - ephemeral         |        ✅        |                    |                      |         ←        |          →         |
| Analysis                  |        ✅        |                    |                      |         ←        |                    |
| Semantic model            |                 |                    |                      |         ←        |          →         |
| Metric                    |                 |                    |                      |         ←        |          →         |
| Saved query               |                 |                    |                      |         ←        |                    |
| Export                    |                 |          ✅         |                      |         ←        |                    |
| Exposure                  |                 |                    |                      |         ←        |                    |
| Data test                 |        ✅        |          ☑️         |           ☑️          |         ←        |                    |
| Unit test                 |                 |                    |                      |         ←        |                    |
| Fixture                   |                 |                    |                      |                  |          →         |
| Group                     |                 |                    |                      |                  |          →         |
| Macro                     |        ✅        |                    |                      |                  |                    |

### Additional information

Implementing this would unlock or enhance the following:
- https://github.com/dbt-labs/docs.getdbt.com/issues/4313
- clarifying updates to `name` within https://docs.getdbt.com/reference/artifacts/manifest-json#resource-details
- clarifying updates to `type` within https://docs.getdbt.com/reference/artifacts/other-artifacts#graph_summaryjson

### PR DESCRIPTION
**Pull request by mirnawong1**
[docs project](https://www.notion.so/dbtlabs/Product-Documentation-a5da199942f94e8da38e9927be3712e3?p=f546559888c041a4ab30b4597043d67c&pm=c)

adding a new page that explains all the available resource types dbt supports and also their supported properties/configs.

Resolves #4314

https://github.com/dbt-labs/docs.getdbt.com/issues/4314

### CODE REVIEW

File: website/docs/reference/about-resources.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+dbt supports different types of resource types. Each one has a set of supported [properties and configurations](/reference/configs-and-properties) within a dbt project, which are key to how they function and integrate within your data project. 
+
+## Resource types table
+
+The following tables describes each resource type, its identifier, and a brief description of its purpose. 
+
+WHERE DO WE LINK TO FOR EACH RESOURCE TYPE? CONTENT UNDER DOCS? OR UNDER RESOURCE CONFIGS?
+
+| Resource type     | Identifier         | Description |
+|-------------------|--------------------|-------------|
```
Comment (mirnawong1) [inline website/docs/reference/about-resources.md]: WHERE DO WE LINK TO FOR EACH RESOURCE TYPE? CONTENT UNDER DOCS? OR UNDER RESOURCE CONFIGS?

File: website/docs/reference/about-resources.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+sidebar_label: "About resource types"
+---
+
+dbt supports different types of resource types. Each one has a set of supported [properties and configurations](/reference/configs-and-properties) within a dbt project, which are key to how they function and integrate within your data project. 
+
+## Resource types table
+
+The following tables describes each resource type, its identifier, and a brief description of its purpose. 
+
+WHERE DO WE LINK TO FOR EACH RESOURCE TYPE? CONTENT UNDER DOCS? OR UNDER RESOURCE CONFIGS?
```
Comment (mirnawong1) [inline website/docs/reference/about-resources.md]:
```suggestion
```

File: website/docs/reference/configs-and-properties.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Configs, properties, what are they?
+title: About configs and properties
```
Comment (mirnawong1) [inline website/docs/reference/configs-and-properties.md]:
```suggestion
title: Configs, properties, what are they?
```

**Comment by matthewshaver** [comment]
@mirnawong1 Any status updates for this draft?

**Comment by mirnawong1** [comment]
hey @matthewshaver , yes still working on through whenever i have a spare moment. it's been a bit busy so haven't had time to write this all up since we have product updates to prioritize

**Pull request closed without merging**